Cython/Compiler/ModuleNode.py

```diff
@@ -3096,7 +3096,15 @@ def generate_module_init_func(self, imported_modules, env, code):
             code.put_xdecref(cname, type)
         code.putln('if (%s) {' % env.module_cname)
         code.putln('if (%s) {' % env.module_dict_cname)
+        # We can run into errors before the stringtab is initialize.
+        # In this case it is not safe to add a traceback (because it uses the stringtab)
```

Comment:
It doesn't globally use the `stringtab`, though, just one specific entry from it: the function name that we pass below.

We currently use `char*` for the function name and file path, which is admittedly suboptimal, but could simplify this code since we don't actually need the `stringtab` for that. I'm unsure, though, because I'd rather like to use prepared Python strings than `char*` in the future.

Reply:
It's actually `__pyx_n_s_cline_in_traceback` in `__Pyx_CLineForTraceback` that it uses (i.e. it's nested pretty deep in the traceback generating code, and I wouldn't like to guarantee that it's the only one).

Reply:
Good to know. Still, my general review comment still applies that there's a cleaner way to do this.